lib/Doctrine/ORM/Internal/Hydration/AbstractHydrator.php

```diff
@@ -414,9 +414,8 @@ protected function gatherRowData(array $data, array &$id, array &$nonemptyCompon
      * values according to their types. The resulting row has the same number
      * of elements as before.
      *
-     * @param array $data
```

Comment:
This removes param information for non-psalm interpreters. I saw that also in older batches.
I you want I can add a PR to fix this as well as using psalm-param / psalm-var for all array<> notations.

Reply:
I'm not 100% sure yet if `psalm-param` or `psalm-var` is the way to go or if we can use just `@param`. I decided to take no risk of providing information that cannot be understood, maybe we should discuss this more in depth? What non-psalm interpreters are you thinking about? PHPStorm, probably, is there something else?

Reply:
I also don't know how well IDEs like eclipse or netbeans are working with array<> for example. At least I hope there are plugins to enable understanding them.
At least we should always have `@param` if a typing is missing.
I prefer removing the psalm prefix for this notation. but there might be more investigation needed to find out how well it is supported in more IDEs.

Reply:
After some investigation my conclusion is that array generic notation should be added to psalm-prefixed PHPDoc attributes only.
The adaption in tools other than intellij, Psalm or PHPStan seems not existent.
[It had been implemented in PHPStorm](https://blog.jetbrains.com/phpstorm/2018/09/phpstorm-2018-3-eap-183-2635-12/#psr-5-generics) according to [PSR-5](https://github.com/php-fig/fig-standards/blob/master/proposed/phpdoc.md)

The PSR-5 chapter for collections had been removed from the draft a month later and did not find a way back yet.
[Here are some more information in the PR removing the collection part](https://github.com/php-fig/fig-standards/pull/1102)
The goal was to newly start the discussion of this topic. I did not try to find out how this proceeded.

Reply:
I feel this is an important discussion we need to have regularly (because tooling will evolve, and new PSRs might come up). I know we have already discussed this in the past somewhere, but cannot find where.

I think there should either be an issue for this at https://github.com/doctrine/coding-standard/issues or a contribution should be done to https://github.com/doctrine/doctrine-website/tree/master/source/policies . Can you pick one and open an issue there? That way we can point Doctrine members and people that are knowledgeable about various IDE support to it.

In any case, my point is that we should have a central location where we record this decision and what it is grounded on so that it can be found and challenged in the future, when tooling has evolved. Having a table with syntaxes and how supported they are across IDEs would help a lot.

We usually use https://github.com/orgs/doctrine/teams/doctrinecore for that kind of thing, but it is internal, so that would prevent us from discussing it with you.

Reply:
I believe it makes sense that we have `@param array` + `@psalm-param array{....}`, probably a phpcs rule that can autofix the param for us, but it needs some work in the coding standard repo.